### PR TITLE
Movable region of interest

### PR DESCRIPTION
**Pull request by Li-Andy**
- Region of Interest Label can now be moved to non-centered position on storyboard 
    - Mask and cropped image (sent to ML model) placement are dynamically set respective to the Region of Interest Label

### CODE REVIEW

File: CardScan/Classes/ScanViewController.swift [hunk 1]
```diff
@@ -484,16 +488,45 @@ import Vision
...
+        let regionOfInterestCenterX = self.regionOfInterestLabelFrame.origin.x + self.regionOfInterestLabelFrame.size.width / 2.0
+        
+        let regionOfInterestCenterY = self.regionOfInterestLabelFrame.origin.y + self.regionOfInterestLabelFrame.size.height / 2.0
+        
+        // calculate center of cropping region in Pixels.
+        var cx, cy: CGFloat
+        
+        // Find out whether left/right or top/bottom of the image was cropped before it was displayed to previewView.
+        // The size of the cropped region is needed to map regionOfInterestCenter to the image center
+        let imageAspectRatio = CGFloat(image.width) / CGFloat(image.height)
```
Comment (kingst) [inline CardScan/Classes/ScanViewController.swift]: Consider adding an assertion to make sure that the previewView is using the scaling style that fits this calculation.

File: Example/CardScan.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -223,7 +223,7 @@
 			isa = PBXProject;
 			attributes = {
 				LastSwiftUpdateCheck = 1020;
-				LastUpgradeCheck = 1000;
+				LastUpgradeCheck = 1020;
```
[the remark this answers is not in the document]
Comment (Li-Andy) [reply Example/CardScan.xcodeproj/project.pbxproj]: - `LastUpgradeCheck` appears to be Xcode's way of checking for Xcode version differences across contributors. 

- CLANG changes are from an Xcode 10.2 prompt suggesting to "Enable Base Internationalization"

> Enabling Base Internationalization is recommended for all projects, and any projects that don’t currently use Base Internationalization are offered an upgrade even if they only have a single localization. The upgraded projects are backward-compatible with previous releases of Xcode.

File: CardScan/Assets/CardScan.storyboard [hunk 3]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <document type="com.apple.InterfaceBuilder3.CocoaTouch.Storyboard.XIB" version="3.0" toolsVersion="14490.70" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES">
-    <device id="retina6_1" orientation="portrait">
```
[the remark this answers is not in the document]
Comment (Li-Andy) [reply CardScan/Assets/CardScan.storyboard]: As we talked about, we'll leave it in this time and omit such changes from future PRs

File: CardScan/Classes/ScanViewController.swift [hunk 4]
```diff
@@ -484,16 +488,45 @@ import Vision
...
+        let screenAspectRatio = screenWidth / screenHeight
+
+        // convert from points to pixels and account for the cropped region
+        if imageAspectRatio > screenAspectRatio {
```
[the remark this answers is not in the document]
Comment (Li-Andy) [reply CardScan/Classes/ScanViewController.swift]: As discussed, added comments to indicate which devices both cases were tested on

File: CardScan/Classes/ScanViewController.swift (hunk 1 above)
Comment (Li-Andy) [reply CardScan/Classes/ScanViewController.swift]: Done.
`assert()` used to avoid crashes in production

**Pull request merged**